lib/Dialect/Seq/SeqOps.cpp

```diff
@@ -57,6 +57,8 @@ ParseResult parseCompRegOp(OpAsmParser &parser, OperationState &result) {
     if (!resultName.empty() && !isdigit(resultName[0]))
       result.addAttribute("name",
```

Comment:
Can you simplify this by doing something like `if !empty && isdigit(resultName[0])) resultName = "";` and fold the two addAttribute calls together?